streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -294,7 +297,13 @@ private void closeDirtyAndRevive(final Collection<Task> taskWithChangelogs, fina
 
                 task.addPartitionsForOffsetReset(assignedToPauseAndReset);
             }
+            if (stateUpdater != null) {
+                tasks.removeTask(task);
```

Comment:
Any particular reason why we remove the task from `tasks` before reviving it? It would seem cleaner to me to basically remove from stream thread / add back to state updater directly in one block, which makes it easier to track ownership.

Reply:
Got it. Could make sense to add a little comment, but I won't insist